tokio/src/runtime/handle.rs

```diff
@@ -201,6 +204,105 @@ impl Handle {
         let _ = self.blocking_spawner.spawn(task, &self);
         handle
     }
+
+    /// Run a future to completion on the Tokio runtime. This is the
+    /// runtime's entry point.
+    ///
+    /// This runs the given future on the runtime, blocking until it is
+    /// complete, and yielding its resolved result. Any tasks or timers
+    /// which the future spawns internally will be executed on the runtime.
+    ///
+    /// # Multi thread scheduler
+    ///
+    /// When the multi thread scheduler is used this will allow futures
+    /// to run within the io driver and timer context of the overall runtime.
+    ///
+    /// # Current thread scheduler
+    ///
+    /// When the current thread scheduler is enabled `block_on`
```

Comment:
While this is the behavior of `Runtime::block_on` I don't think this should be the behavior of `Handle::block_on`? @carllerche What do you think? I don't think the handle's block on should be able to steal the runtime. In that case it is the same as `Runtime` except it removes the ownership of the runtime type.

Reply:
Well, my initial thinking was that the handle should follow the Runtime's behavior, but it does complicate the shutdown logic as well as seems a bit weird given the "current_thread" name... Perhaps, `block_on` in this case should just block the thread and enter the runtime context...